smart-contracts/test/hook.js

```diff
@@ -0,0 +1,125 @@
+const { ethers } = require('@nomiclabs/buidler')
+const { BigNumber, constants, utils } = require('ethers')
+const {
+  deployLock,
+  deployHook,
+  hookAddress,
+  deployToken,
+} = require('./fixtures.js')
+const hookJSON = require('../artifacts/BondingCurveHook.json')
+const hookABI = hookJSON.abi
+const { assert } = require('chai')
+const provider = ethers.provider
+const DENOMINATOR = Math.pow(2, 64)
+const CURVE_MODIFIER = 3.321
+const ZERO_ADDRESS = utils.getAddress(
+  '0x0000000000000000000000000000000000000000'
+)
+let walletAddress
+
+function jsPriceCalculator(s) {
+  return Math.log2(s) / CURVE_MODIFIER
+}
+
+function fixedPointToDecimal(int128Numerator) {
+  return int128Numerator / DENOMINATOR
+}
+
+let hook
+let lockedFyiLock
+let purchaseHook
+let tokenAddress
+let deployedHookAddress
+
+describe('Lock Setup', () => {
+  before(async () => {
+    const [wallet] = await ethers.getSigners()
+    // Get the deployed lock:
+    const results = await deployLock()
+    lockedFyiLock = results[0]
+    tokenAddress = results[1]
+
+    // Get the deployed hook:
+    if (hookAddress === undefined) {
+      purchaseHook = await deployHook(lockedFyiLock.address)
+    } else {
+      purchaseHook = await ethers.getContractAt(hookABI, hookAddress)
+    }
+
+    // Ensure we're using the correct signer:
+    walletAddress = await wallet.getAddress()
+    assert.isOk(await lockedFyiLock.isLockManager(walletAddress))
+
+    //Register the purchase hook:
+    const receipt = await lockedFyiLock.setEventHooks(
+      purchaseHook.address,
+      ZERO_ADDRESS //constants.ZeroAddress,
+    )
+    const returnedHookAddress = await lockedFyiLock.onKeyPurchaseHook()
+    assert.equal(returnedHookAddress, purchaseHook.address)
+
+    // Set the beneficiary to the Locked-FYI-DAO address:
+    // await lockedFyiLock.updateBeneficiary()
+  })
```

Comment:
I'm thinking to extract all of this to `setup.js` and export a `setup()` function to be called here in the `before` func.